routers/repo/issue.go

```diff
@@ -232,8 +232,15 @@ func issues(ctx *context.Context, milestoneID int64, isPullOption util.OptionalB
 				ctx.ServerError("LoadPullRequest", err)
 				return
 			}
+			pull := issues[i].PullRequest
 
-			commitStatus[issues[i].PullRequest.ID], _ = issues[i].PullRequest.GetLastCommitStatus()
+			if err := pull.LoadProtectedBranch(); err == nil {
```

Comment:
You're not handling the error here. Please do something like
```
if err := pull.LoadProtectedBranch(); err != nil {
    return
}

if pull.ProtectedBranch != nil && pull.ProtectedBranch.RequiredApprovals != 0 {
    ProtectedBranch.GrantedApprovalsCount = pull.ProtectedBranch.GetGrantedApprovalsCount(
}
```

Reply:
Aah ok, is there a subtle difference in the behaviour of errors and nil?

Reply:
If the function returns `nil` for an error, there was no error. You don't check for the case where the error was not `nil` which means you're never actually doing something with the error.

In Go, there is this pattern of keeping a "happy path" of code, that's why you'll see a lot of `if err != nil ` code used to handle errors. Take a look at this article for more information: https://medium.com/@matryer/line-of-sight-in-code-186dd7cdea88